boost-maven-plugin/src/main/java/io/openliberty/boost/liberty/LibertyPackageMojo.java

```diff
@@ -65,46 +65,83 @@ public void execute() throws MojoExecutionException {
         createDefaultRuntimeArtifactIfNeeded();
 
         createServer();
-        
+
         /**
          * Whether the packaged Liberty Uber JAR will be the project artifact. This
          * should be the case in Spring Boot scenarios since Spring Boot developers
-         * expect a runnable JAR. 
+         * expect a runnable JAR.
          */
         boolean attach;
 
+        /**
+         * Use the classifier to determine whether we need to set the Liberty Uber JAR
+         * as the project artifact, and add Spring-Boot-Version to the manifest
+         */
+        String springBootClassifier = null;
+
         try {
-            if (springBootVersion != null && !springBootVersion.isEmpty()) {
-                // Dealing with a spring boot app
-                attach = true;
-                validateSpringBootUberJAR();
-                copySpringBootUberJar();
+            if (MavenProjectUtil.isValid(springBootVersion)) { // Dealing with a spring boot app
+                springBootClassifier = net.wasdev.wlp.maven.plugins.utils.SpringBootUtil
+                        .getSpringBootMavenPluginClassifier(project, getLog());
+
+                // Check if we need to attach based on the classifier configuration
+                if (MavenProjectUtil.isValid(springBootClassifier)) {
+                    attach = false;
+                } else {
+                    attach = true;
+                }
+
+                File springBootUberJar = net.wasdev.wlp.maven.plugins.utils.SpringBootUtil.getSpringBootUberJAR(project,
+                        getLog());
+                validateSpringBootUberJAR(springBootUberJar);
+                copySpringBootUberJar(springBootUberJar, attach); // Only copy back if we need to overwrite the project
+                                                                  // artifact
                 installApp("spring-boot-project");
                 generateServerXML();
                 generateBootstrapProps();
-            } else {
-                // Dealing with an EE based app
+                installMissingFeatures();
+                
+                if(springBootUberJar != null) {
+                    createUberJar(springBootUberJar.getAbsolutePath(), attach);
+                }
+                else {
+                    createUberJar(null, attach);
+                }
+                
+                if (!MavenProjectUtil.isValid(springBootClassifier)) {
+                    // If necessary, add the manifest to prevent Spring Boot from repackaging again
+                    addSpringBootVersionToManifest(springBootVersion);
```

Comment:
Hmm... I don't really have a strong reason, but my guess would have been to just add the manifest entry anyway to the Liberty Uber JAR.    That is:   "any time Boost packages a Liberty UBer JAR from SpringBoot app you see this manifest entry" rather than only adding it because we have a reason based on the packaging sequence.  That kind of consistency strikes me as better.

Reply:
While this is possible, it adds more technical complexity since we then need a way of identifying the Liberty Uber JAR in both the classifier and non-classifier scenarios. It's not as simple as just getting the project artifact, which is what we currently do. Plus there's no additional value to the consistency, since we have other methods of identifying a Liberty Uber JAR (even in Spring Boot thin scenarios).